### PR TITLE
🔥 726 - remove confirm password

### PR DESCRIPTION
**Pull request by anncatton**
## Description of Changes

Removes "confirm password" requirement from Registration flow, no longer needed with granular password errors implemented in #718 .
Note: Includes some changes to allow whitespace in passwords.

PR # 3 of Password changes for #718 + #726 

### Consent UI
- removes `confirmPassword` input and logic from Registration form
- removes related translation keys
 
**For password whitespace changes**: 
- removes references to `invalidPasswordCharacter` in validation messages, translations

### Types
- removes `confirmPassword` from `Password` schema
- removes `hasMatchingPasswords` function + related tests
- removes references from `participantRegistration` tests
- moves `hasRequiredPasswordElements` function to `password.ts` util file

**For password whitespace changes**:
- updates regexes
- removes `invalidPasswordCharacter` check in `hasRequiredPasswordElements`
- updates tests  

### Special Instructions
Pick up types changes by running `pnpm run build` from `packages/types` folder

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [x] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [ ] Added copyrights to new files

### CODE REVIEW

File: packages/types/tests/entities/Password.spec.ts [hunk 1]
```diff
@@ -74,11 +73,11 @@ describe('Password', () => {
 			expect(Password.safeParse('LDpFymuoDGz.&!hIuS').success).true;
 		});
 
-		it('Does not validate if it contains whitespace', () => {
-			expect(Password.safeParse('lhmOe6l85Pw2 jBQ1HfO').success).false;
-			expect(Password.safeParse('lhm    Oe6l85Pw26H jBQ1HfO').success).false;
-			expect(Password.safeParse('lhmOe6l85Pw26HjBQ1HfO  ').success).false;
-			expect(Password.safeParse('  lhmOe6l85Pw26HjBQ1HfO').success).false;
+		it('Validates if it contains whitespace', () => {
```
Comment (mistryrn) [inline packages/types/tests/entities/Password.spec.ts:76]: 🎉

**Review by mistryrn — APPROVED** [review]
Looks great! Thanks for updating this so quickly 🎉

File: packages/types/src/services/consentUi/requests/Register.ts [hunk 2]
```diff
@@ -83,49 +77,14 @@ const RegisterEmailAddressFieldsRefined = RegisterRequestEmailAddressFields.supe
...
-	.refine(hasMatchingPasswords, {
-		// TODO: chaining refine() call to superRefine() will be removed in https://github.com/OHCRN/platform/issues/726
-		message: 'passwordMismatch',
-		path: ['confirmPassword'],
-	});
+	.superRefine(hasRequiredPasswordElements);
 
 export const RegisterFormStep2 = RegisterEmailAddressFieldsRefined.and(ConsentToBeContacted)
-	.and(PasswordFields)
+	.and(PasswordField)
```
Comment (samrichca) [inline packages/types/src/services/consentUi/requests/Register.ts:87]: 👍

Comment (samrichca) [inline packages/types/src/services/consentUi/requests/Register.ts:87]: 👍

**Review by samrichca — APPROVED** [review]
looks good!

**Pull request merged**